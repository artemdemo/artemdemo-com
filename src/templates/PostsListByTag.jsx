import React from 'react';
import _get from 'lodash/get';
import {graphql} from 'gatsby';
import PostsList from '../pages/PostsList/PostsList';
import * as utils from '../services/utils';

class PostsListByTag extends PostsList {
  constructor(props) {
    super(props);
  }

  renderBeforeContent() {
    const tagName = _get(this.props, 'pageContext.tagName', '')
    return (
      <p>
        <strong>Tag:</strong>
        &nbsp;&nbsp;{tagName}
      </p>
    );
  }

  renderPagination() {
    const tagName = _get(this.props, 'pageContext.tagName', '');
    return super.renderPagination(utils.prefixTag.bind(null, tagName))
  }
}

export default PostsListByTag;

// Notice, that this query should be the same as in `/templates/PostsList.jsx`
// ToDo: Find the way to keep both queries in one place
export const pageQuery = graphql`
  query($tagName: String, $skip: Int!, $limit: Int!) {
    site {
      siteMetadata {
        title
        description
      }
    }
    allMarkdownRemark(
      sort: { fields: [frontmatter___date], order: DESC }
      skip: $skip
      limit: $limit
      filter: { frontmatter: { tags: { in: [$tagName] } } }
    ) {
      totalCount
      edges {
        node {
          excerpt(pruneLength: 500)
          fields {
            slug
          }
          frontmatter {
            date(formatString: "DD MMMM, YYYY")
            title
          }
        }
      }
    }
  }
`;
