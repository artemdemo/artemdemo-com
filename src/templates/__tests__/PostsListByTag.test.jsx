import React from 'react';
import renderer from 'react-test-renderer';
import PostsListByTag from '../PostsListByTag';

jest.mock('../../components/Posts/Posts');
jest.mock('../../components/Pagination/Pagination');
jest.mock('../../components/Layout/Layout');
jest.mock('../../components/Container/Container');

describe('PostsListByTag', () => {
    it('default render',() => {
        const tree = renderer.create(
            <PostsListByTag />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with tagName',() => {
        const tree = renderer.create(
            <PostsListByTag
                pageContext={{
                    tagName: 'Tag name',
                }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});