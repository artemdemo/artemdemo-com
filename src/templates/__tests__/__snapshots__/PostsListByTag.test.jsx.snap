// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsListByTag default render 1`] = `
<div
  data-mock="Container"
>
  {}
  <p>
    <strong>
      Tag:
    </strong>
      
    
  </p>
  <div
    data-mock="Posts"
  >
    {}
  </div>
  <div
    data-mock="Pagination"
  >
    {
  "skip": 0,
  "limit": 0,
  "totalPosts": 0
}
  </div>
</div>
`;

exports[`PostsListByTag should render with tagName 1`] = `
<div
  data-mock="Container"
>
  {}
  <p>
    <strong>
      Tag:
    </strong>
      
    Tag name
  </p>
  <div
    data-mock="Posts"
  >
    {}
  </div>
  <div
    data-mock="Pagination"
  >
    {
  "skip": 0,
  "limit": 0,
  "totalPosts": 0
}
  </div>
</div>
`;
